src/amltk/exceptions.py

```diff
@@ -110,3 +110,50 @@ def __str__(self) -> str:
             f"Duplicate names found in {self.node.name} and can't be handled."
             f"\nnodes: {[n.name for n in self.node.nodes]}."
         )
+
+
+class MatchDimensionsError(Exception):
```

Comment:
I see that they're kind of used as KeyError's mostly, can we inherit from that instead?